codec/helper.go

```diff
@@ -309,6 +311,13 @@ type Selfer interface {
 	CodecDecodeSelf(*Decoder)
 }
 
+// UnknownFieldsHandler defines methods by which a value can store
+// unknown fields encountered during decoding.
+type UnknownFieldsHandler interface {
+	CodecOnUnknownField(fieldName string, fieldValue interface{})
```

Comment:
Note to self: it might be possible to avoid decoding the value at all and just pass the encoded bytes here.